### PR TITLE
Feature/rework orizuru

### PR DESCRIPTION
**Pull request by mwoodffdc**
- Rework of Orizuru

### CODE REVIEW

File: src/example/web.js [hunk 1]
```diff
@@ -55,13 +55,35 @@ const
 	}],
 
 	// define port (should be an env var in production)
-	port = 5555;
+	PORT = 5555;
```
Comment (dfruddffdc) [inline src/example/web.js]: Constant should be at the top, after requires

File: src/lib/index/server.js [hunk 2]
```diff
@@ -25,236 +25,211 @@
...
+			me = this,
+			schema = config.schema,
+			responseWriter = config.responseWriter,
+			fullSchemaName = schema.name,
+			schemaNameParts = fullSchemaName.split('.'),
+			schemaNamespace = _.initial(schemaNameParts).join('.'),
+			schemaName = _.last(schemaNameParts),
+			apiEndpoint = `${config.endpoint}${schemaNamespace}`.replace(/\./g, '/'),
+
+			routeConfiguration = _.get(me[PROPERTY_ROUTE_CONFIGURATION], apiEndpoint, {});
```
Comment (dfruddffdc) [inline src/lib/index/server.js]: Works fine as is, but you're effectively splitting the path out over the first and second args.
I'd probably do
```javascript
_.get(me, `${PROPERTY_ROUTE_CONFIGURATION}.apiEndpoint`, {})
```

File: src/example/web.js (hunk 1 above)
Comment (mwoodffdc) [reply src/example/web.js]: Done

File: src/lib/index/server.js (hunk 2 above)
Comment (mwoodffdc) [reply src/lib/index/server.js]: Done

File: src/spec/index/publisher.spec.js [hunk 3]
```diff
@@ -119,6 +121,11 @@ describe('index/publisher.js', () => {
 					message: {
 						first: 'First',
 						last: 'Last'
+					},
+					context: {
+						user: {
+							username: 'test@test.com'
```
Comment (dfruddffdc) [inline src/spec/index/publisher.spec.js]: `test.com` is a real business domain, better to use `example.com`
But I appreciate test.com has already been used all over the codebase already, and when in rome...

File: src/lib/index/server.js [hunk 4]
```diff
@@ -25,236 +25,212 @@
...
+			// Validate the config
+			new ServerValidator(config);
 
-		// add server to private store
-		serverStore[this] = server;
+			// Create the express server
+			const server = express();
 
-	}
+			// Configure the server
```
Comment (sfryffdc) [inline src/lib/index/server.js]: We should not be adding any middleware here. These could be added to the routes added but probably better to be completely un opinionated. The problem is that this middleware will go off first on all requests. So DOS attacks sending huge payloads could be crafted and bodyparser will be very busy.

Comment (sfryffdc) [reply src/lib/index/server.js]: It seems that express now has a json middleware built in too.

**Comment by bslaterffdc** [comment]
@mwoodffdc @sfryffdc This looks like a step in the right direction.
If it contains breaking API changes, can we document what they are somewhere?
Don't mind migrating, but don't want to have to read the code (or docs really) to work out how.
A small summary somewhere would go a long way, like we do with fflib release notes, etc.

File: src/lib/index/server.js (hunk 4 above)
Comment (mwoodffdc) [reply src/lib/index/server.js]: Done

**Pull request merged**